airbyte-integrations/connectors/source-mixpanel/source_mixpanel/components.py

```diff
@@ -337,3 +342,31 @@ def get_json_schema(self) -> Mapping[str, Any]:
                 schema["properties"][property_name] = types.get(property_type, {"type": ["null", "string"]})
         self.schema = schema
         return schema
+
+
+class CohortMembersLegacyToPerPartitionStateMigration(LegacyToPerPartitionStateMigration):
+    """Cohort Members stream should not support incremental syncs because
+    the only possible cursor 'last_seen' is not suitable for this purposes
+    since members of cohorts can be changed any time.
+
+    So here, just reset legacy state
+    """
+
+    config: Mapping[str, Any]
+    parameters: Mapping[str, Any]
+
+    def __init__(
+        self,
+        config: Mapping[str, Any],
+        parameters: Mapping[str, Any],
+    ):
+        self._config = config
+        self._parameters = parameters
+
+    def should_migrate(self, stream_state: Mapping[str, Any]) -> bool:
+        if _is_already_migrated(stream_state):
+            return False
+        return True
+
+    def migrate(self, stream_state: Mapping[str, Any]) -> Mapping[str, Any]:
+        return {"states": []}
```

Comment:
Why we need reset state?

Reply:
Do you mean that state reset is not needed because after connector update  data for cohort_members stream will be reset?

Reply:
I have removed custom state migration